src/reshape.jl

```diff
@@ -0,0 +1,29 @@
+import IndexedTables: stack, unstack
+export stack, unstack
+
+function stack(t::DDataset, by = pkeynames(t); select = isa(t, DNDSparse) ? valuenames(t) : excludecols(t, by),
+    variable = :variable, value = :value)
+
+    function stackchunk(x)
+        stack(x, by; select=select, variable=variable, value=value)
+    end
+
+    return fromchunks(delayedmap(stackchunk, t.chunks))
+end
+
+function unstack(::Type{D}, ::Type{T}, key, val, cols::AbstractVector{S}) where {D <:DDataset, T, S}
+    D1 = D isa DNDSparse ? NDSparse : NextTable
+    function unstackchunk(x, y)
+        unstack(D1, T, x, y, cols)
+    end
+    fromchunks(delayedmap(unstackchunk, key.chunks, val.chunks))
+end
+
+function unstack(t::D, by = pkeynames(t); variable = :variable, value = :value) where {D<:DDataset}
+    tgrp = groupby((value => identity,), t, by, select = (variable, value))
```

Comment:
What does this groupby do? Why do you group by primary key..?

Reply:
`unstack` needs some indexing column to put together values that will be assigned to the same row in the unstacked version. The easiest way I could think of is in two steps:

- Group all those values in a small table (cheap, as I'm just doing views)
- For each of these small tables, expand it to a row (leaving `NA` if it's missing from the table)

The default indexing column(s) in `unstack` are the primary keys now, which works very nicely as it gives `unstack(stack(t)) == t`.

`stack` also requires indexing columns now, though that isn't strictly required in order to compute it, maybe I could allow `by == ()` in `stack`

Reply:
Ok it makes sense, I think. Are there cases where this wouldn't work?

Reply:
I don't think so because the `groupby` makes sure all the different rows to be combined together end up on the same processor, which is the only tricky bit of the parallel implementation.